rclcpp/src/rclcpp/executors/static_single_threaded_executor.cpp

```diff
@@ -105,9 +101,21 @@ StaticSingleThreadedExecutor::spin_some_impl(std::chrono::nanoseconds max_durati
 
   while (rclcpp::ok(context_) && spinning.load() && max_duration_not_elapsed()) {
     // Get executables that are ready now
-    entities_collector_->refresh_wait_set(std::chrono::milliseconds::zero());
+    std::lock_guard<std::mutex> guard(mutex_);
+    if (current_collection_.empty() || this->entities_need_rebuild_.load()) {
+      this->collect_entities();
+    }
+
+    auto wait_result = wait_set_.wait(std::chrono::nanoseconds(0));
+    if (wait_result.kind() == WaitResultKind::Empty) {
+      RCUTILS_LOG_WARN_NAMED(
+        "rclcpp",
+        "empty wait set received in wait(). This should never happen.");
+      continue;
+    }
```

Comment:
There are 3 identical copies of this code here, perhaps put it into a function ?

Reply:
Refactored into `collect_and_wait`

Reply:
looks good to me, can I somehow mark things as resolved ? Gitlab has this functionality, but I can't find it on github....

Reply:
Do you see the "resolve conversation" button.  Maybe a permissions thing.

Reply:
Nope, no button...